src/logic/proposals/encoding/encoding.ts

```diff
@@ -112,6 +117,7 @@ export const goEncodeEvmScript = async (
       range(parameterTypes.length).map(async (i) => {
         const param = targetParameters[i]!;
         if (parameterTypes[i] !== 'address') return param;
+        if (typeof param !== 'string') throw new Error('Parameter must be an ENS string');
```

Comment:
The error message `Parameter must be an ENS string` makes it sound like we only accept ENS names here.

Reply:
Mhh, but it will be only present if there is a non-numeric value passed as address. But I guess it doesn't hurt to reword this.

Reply:
Actually, I remember why I didn't put much effort into this error message. This was mainly for type inference to pass string to `convertToAddressOrThrow`. Even if this throws we still use the generic `Ensure parameters match target contract signature` message. I think this is good enough, but I've added a test for this. If you disagree open an issue (it don't wanna implement it in this PR).